File: 장세훈/LeetCode/ValidPalindrome.java
package 장세훈.LeetCode;

public class ValidPalindrome {

    /*
        125. Valid Palindrome
        https://leetcode.com/problems/valid-palindrome/description/

        입련된 문자열을 정규식으로 알파벳과 숫자만 필터링하고 알파벳은 소문자로 만든다.
        변환된 문자열 기준으로 앞뒤 문자가 같은지 확인하며 순회한다.
     */
    public boolean isPalindrome(String s) {
        String str = s.replaceAll("[^A-Za-z0-9]", "").toLowerCase();

        for (int i = 0; i < str.length() / 2; i++) {
            if (str.charAt(i) != str.charAt(str.length() - 1 - i)) {
                return false;
            }
        }

        return true;
    }

}

File: 정수빈/LeetCode/top_interview/valid-palindrome.py
# https://leetcode.com/problems/valid-palindrome/
# 시간: 4분 30초

class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        s = s.lower()
        new_s = ""
        for c in s:
            if str.isalpha(str(c)) or str.isdigit(str(c)):
                new_s += c
        lp = 0
        rp = len(new_s) - 1
        while lp <= rp:
            if new_s[lp] != new_s[rp]:
                return False
            lp += 1
            rp -= 1
        return True
        